apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java

```diff
@@ -410,6 +410,11 @@ public CorrelationContext getCorrelationContext() {
         return this.correlationContext;
     }
 
+    @Override
+    public String getPrimaryEndpointName() {
+        return (primaryEndpoint != null) ? primaryEndpoint.getName() : null;
```

Comment:
**primaryEndpoint** should not be NULL in any case. I don't think we need null-check.

Reply:
What about if it's called before the first push?

https://github.com/apache/skywalking-java/blob/bd7912106aede5f79dbcfcdafe5c13c4378d8629/apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/TracingContext.java#L529

Reply:
How a tracing context existing without a span created? I think that would cause more issues.

Reply:
Take a deep look at how this method is being called, `ContextManager` can't create a TracingContext(AbstractTracerContext) without creating a span.

Reply:
Looked into it further. You're right. Removed null check.